aiohttp/web_app.py

```diff
@@ -517,29 +517,23 @@ async def _handle(self, request: Request) -> StreamResponse:
 
         match_info.freeze()
 
-        resp = None
         request._match_info = match_info
         expect = request.headers.get(hdrs.EXPECT)
         if expect:
-            resp = await match_info.expect_handler(request)
+            await match_info.expect_handler(request)
```

Comment:
This returns `None`. No idea if this functionality actually did something in the past, but appears to be a completely redundant check today.

Reply:
Taking another look at the documentation, maybe it should be allowed to return a response. In which case, there should be a test case to cover this:
> If handler returns an instance of class [StreamResponse](https://docs.aiohttp.org/en/stable/web_reference.html#aiohttp.web.StreamResponse), request handler uses it as response.
https://docs.aiohttp.org/en/stable/web_advanced.html?highlight=expect#expect-header

Reply:
https://github.com/aio-libs/aiohttp/pull/7025

Reply:
OK, definitely doesn't work. We can either drop the functionality as done here, or merge #7025 with the fix, then I'll update this.